Adding IRSA + Upgrade cert-manager version (#8009)
solo-io/gloo: docs/content/guides/integrations/cert_manager/_index.md

```shell
kubectl create namespace cert-manager
kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.11.0/cert-manager.yaml
```

### Add a Service

> Add a service that will get exposed via Gloo Edge. In this document we will use the petclinic spring application. 

Reviewer: ```suggestion
Deploy the following `petclinic` Spring application to expose in Gloo Edge.
```

```shell
kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo/v0.8.4/example/petclinic/petclinic.yaml
```

---

## Utilizing the ACME DNS-01 Challenge

> We'll need to allow cert manager access to configure DNS records in AWS.

Reviewer: ```suggestion
Start by allowing cert manager to configure DNS records in AWS.
```

> This documentation describe the use of an **AWS key pair** for test and development purposes, and it also shows the use of **IAM roles for service accounts (IRSA)** for production use.

Reviewer: ```suggestion
Choose between the following options:
* Development and testing environments: Use an [**AWS key pair**](#using-an-aws-key-pair).
* Production environments: Use [**IAM roles for service accounts (IRSA)**](#using-aws-irsa).
```

See cert manager [docs](https://cert-manager.io/docs/configuration/acme/dns01/route53/) for more details on the access requirements for cert-manager, especially for cross-account case not covered in this page. 

In this example we used the domain name `test-123456789.solo.io`. We'll create an `A` record that maps to the IP address of the 
gateway proxy that we installed with Gloo Edge.

You can run these commands to update AWS route53 through the AWS command line tool 
(remember to replace *HOSTED_ZONE* and *RECORD* with your values):

```shell
export GLOO_HOST=$(kubectl get svc -l gloo=gateway-proxy -n gloo-system -o 'jsonpath={.items[0].status.loadBalancer.ingress[0].ip}')
export RECORD=test-123456789
export HOSTED_ZONE=solo.io.
export ROUTE53_ZONE_ID=$(aws route53 list-hosted-zones|jq -r '.HostedZones[]|select(.Name == "'"$HOSTED_ZONE"'").Id')
export RS='{ "Changes": [{"Action": "UPSERT", "ResourceRecordSet":{"ResourceRecords":[{"Value": "'$GLOO_HOST'"}],"Type": "A","Name": "'$RECORD.$HOSTED_ZONE'","TTL": 300} } ]}'
aws route53 change-resource-record-sets --hosted-zone-id $ROUTE53_ZONE_ID --change-batch "$RS"
```

> ### Using AWS key pair

Reviewer: ```suggestion
### Using an AWS key pair
```

#### Provide AWS account details to cert-manager

> We will need to add the access keys as a kubernetes secret, so that cert-manager can access them:

Reviewer: ```suggestion
Add the access keys as a Kubernetes secret, so that cert-manager can access them:
```

EOF
```

### Using AWS IRSA

> A more production approach should use IAM roles for service accounts (IRSA).
> 
> Make sure it is enable in your EKS cluster before going throught this tutorial. See more on [AWS website](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html).

Reviewer: ```suggestion
For production-level setups, use IAM roles for service accounts (IRSA) to give cert manager the necessary access.

**Before you begin**: Make sure that IRSA is enabled in your EKS cluster. For more information, see the [AWS IRSA documentation](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html).
```

#### Create role and policy

> For convenience, let's create some variables :

Reviewer: ```suggestion
For convenience, save the following information in environment variables:
```

```shell
export AWS_ACCOUNT=$(aws sts get-caller-identity --query Account | tr -d '"')
export EKS_CLUSTER_NAME=my-eks-cluster-name
export EKS_REGION=us-east-1
export HOSTED_ZONE=solo.io.
export ROUTE53_ZONE_ID=$(aws route53 list-hosted-zones|jq -r '.HostedZones[]|select(.Name == "'"$HOSTED_ZONE"'").Id')
export EKS_HASH=$(aws eks describe-cluster --name ${EKS_CLUSTER_NAME} --query cluster.identity.oidc.issuer | cut -d '/' -f5 | tr -d '"')
```

> Now, we can create a policy with the minimum rights needed and attach it to a role :

Reviewer: ```suggestion
Create a policy with the minimum-required access.
```

```shell
cat <<EOF > policy.json
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Action": "route53:GetChange",
      "Resource": "arn:aws:route53:::change/*"
    },
    {
      "Effect": "Allow",
      "Action": [
        "route53:ChangeResourceRecordSets",
        "route53:ListResourceRecordSets"
      ],
      "Resource": "arn:aws:route53:::hostedzone/*"
    },
    {
      "Effect": "Allow",
      "Action": "route53:ListHostedZonesByName",
      "Resource": "*"
    }
  ]
}
EOF

aws iam create-policy \
    --policy-name AwsCertManagerToRoute53 \
    --policy-document file://policy.json
```

> And attach this policy to a role :

Reviewer: ```suggestion
And attach this policy to a role.
```

```shell
cat <<EOF > trust-policy.json
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Action": "sts:AssumeRoleWithWebIdentity",
      "Principal": {
        "Federated": "arn:aws:iam::${AWS_ACCOUNT}:oidc-provider/oidc.eks.${EKS_REGION}.amazonaws.com/id/${EKS_HASH}"
      },
      "Condition": {
        "StringEquals": {
          "oidc.eks.${EKS_REGION}.amazonaws.com/id/${EKS_HASH}:sub": "system:serviceaccount:cert-manager:cert-manager"
        }
      }
    }
  ]
}
EOF

aws iam create-role --role-name EksCertManagerRole --assume-role-policy-document file://trust-policy.json
aws iam attach-role-policy --policy-arn arn:aws:iam::${AWS_ACCOUNT}:policy/AwsCertManagerToRoute53 --role-name EksCertManagerRole

export IAM_ROLE_ARN=$(aws iam get-role --role-name EksCertManagerRole --query Role.Arn | tr -d '"')
```

> The cert-manager service account can then be annotated to use this role to manage route53 records :

Reviewer: ```suggestion
Annotate the `cert-manager` service account to use this role to manage `route53` records.
```

```bash
kubectl annotate sa -n cert-manager cert-manager "eks.amazonaws.com/role-arn"="${IAM_ROLE_ARN}"
```

> To be able to read the ServiceAccount token, it is necessary to modify the cert-manager deployment to define new file system permissions (it can also be done using cert-manager helm chart).

Reviewer: ```suggestion
To enable the cert-manager deployment to read the ServiceAccount token, modify the cert-manager deployment to define new file system permissions with the following command. You can also make these changes by upgrading the Helm chart that you used to deployed cert-manager, which persists the changes across upgrades.
```
